### DISCUSSION TITLE
MASVS-STORAGE Refactoring (till 15.03.22)

### DISCUSSION THREAD

**Discussion by cpholguera** [OWASP/masvs#627] [Big MASVS Refactoring]
Hello everybody, 

as part of the refactoring process we decided to publish our draft of every section of the MASVS that we (@cpholguera, @TheDauntless and @sushi2k) worked on.

This is based on the MASVS category "V2: Data Storage and Privacy Requirements" (from the MASVS Version 1.4.2): <https://github.com/OWASP/owasp-masvs/blob/v1.4.2/Document/0x07-V2-Data_Storage_and_Privacy_requirements.md>

Here you can find a summary of the proposed new requirements (more details below):

<img width="1491" alt="MASVS-STORAGE Proposal" src="https://user-images.githubusercontent.com/29175115/154018165-2517479f-f436-4475-aea7-1b5d5b4143f7.png">

In the following link we include a nice visualization as a _diff_ spreadsheet including:

- Reason for removal/change.
- Idea behind the new requirement/formulation.
- More context thanks to related Test Cases.
- Filter in column F1 to see 
  - the previous version only
  - the new version only
  - or the diff

> **MASVS-STORAGE is open for comments till mid of March 2022. Please comment directly in the Google Spreadsheet (you can comment on each of the cells) or in this GitHub Discussion.**

[<img width="1471" alt="image" src="https://user-images.githubusercontent.com/29175115/154018963-63d119b0-2b64-43ef-847d-51689fdf9a92.png">](https://docs.google.com/spreadsheets/d/1-mddga8DtFu5Ef9vcyPNrgaFMx44Bv5AcPv4ZqFtTHM/edit?usp=sharing)

[MASVS-STORAGE Refactoring Diff](https://docs.google.com/spreadsheets/d/1-mddga8DtFu5Ef9vcyPNrgaFMx44Bv5AcPv4ZqFtTHM/edit?usp=sharing)

**Comment by jmanico** [comment]
I like this change from the current version, nice!

**Comment by cpholguera** [comment]
Closed for comments. Thanks everyone for your feedback so far!

**Comment by cpholguera** [comment]
Consider https://github.com/OWASP/owasp-masvs/discussions/647
